app/services/fastly_vcl/whitelisted_params.rb

```diff
@@ -0,0 +1,65 @@
+module FastlyVCL
+  # Handles updates to our VCL snippet on Fastly that whitelists params
+  class WhitelistedParams
+    VCL_DELIMITER_START = "^(".freeze
+    VCL_DELIMITER_END = ")$".freeze
+    SNIPPET_NAME = "Whitelist certain querystring parameters".freeze
+    FILE_PARAMS = YAML.load_file("config/fastly/whitelisted_params.yml").sort.freeze
+
+    class << self
+      def update
+        fastly = Fastly.new(api_key: ApplicationConfig["FASTLY_API_KEY"])
+        service = fastly.get_service(ApplicationConfig["FASTLY_SERVICE_ID"])
+        latest_version = service.version
+
+        snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                     latest_version.number,
+                                     SNIPPET_NAME)
+
+        current_params = params_to_sorted_array(snippet.content)
+
+        return if current_params == FILE_PARAMS # No update needed
+
+        new_version = latest_version.clone
+        new_snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                         new_version.number,
+                                         SNIPPET_NAME)
+        new_snippet.content = build_content(FILE_PARAMS, new_snippet.content)
+        new_snippet.save!
+
+        new_version.activate!
+        Rails.logger.info("Fastly updated to version #{new_version.number}.")
```

Comment:
maybe a bit more explicit? `Fastly VCL configuration updated...`

Reply:
I wonder if we should also add a metric to Datadog, that can answer instantly the question: "which version of Fastly VCL is in production?", especially because very few people have access to Fastly.

cc @mstruve

Reply:
That would actually be something different. We update the configuration on an entirely new version of Fastly with `new_version.activate!`. The VCL configuration version number is something different entirely. We don't ever adjust that intentionally.

Reply:
And as for Datadog, that's why I logged `"new_version:#{new_version.number}"` to answer that same question "which version of Fastly VCL is in production". We tie VCLs to versions. So if you update the VCL, the version of Fastly is updated. And the version of Fastly that is activated is how we know which VCL is there too. We can't update the VCL without making a new version of Fastly.

Reply:
Awesome, thanks for the explanation :D

Reply:
Leaving this as a note for when I inevitably look back...Fastly does have a concept of [dynamic scripts](https://docs.fastly.com/vcl/vcl-snippets/using-dynamic-vcl-snippets/) which are versionless objects that you can modify independently. I actually tried to change our VCL to be a dynamic script and go that route when I started this.

But 1) the `fastly` gem doesn't fully support them yet and you can only interact with dynamic scripts via the API. So in other words, we'd have to make our own calls/build a wrapper until the gem was updated. And 2) dynamic scripts still required us to update the Fastly version, which was what I was really hoping to avoid by going dynamic in the first place.